### DISCUSSION TITLE
A better way to write this templated ignore?

### DISCUSSION THREAD

**Discussion by VorpalBlade** [twpayne/chezmoi#3547] [Q&A] [answered]
In my bin directory I have a this fragement:

```
{{- if and (ne .chezmoi.osRelease.id "debian") (ne .chezmoi.osRelease.id "raspbian") (ne .chezmoi.osRelease.id "ubuntu") (ne .chezmoi.os "android") }}
debdiffconf
{{- end }}
```

This excludes a dpkg/apt specific script on anything that isn't Debian based (and that I use). This obviously doesn't even work on android/Termux where osRelease is empty, so in fact I have a custom variable and it reads like this instead:

```
{{- if and (ne .custom.linux_flavour "debian") (ne .custom.linux_flavour "raspbian") (ne .custom.linux_flavour "ubuntu") (ne .custom.linux_flavour "android") }}
debdiffconf
{{- end }}
```

What I would *want* to write is something like:

```
{{- if not (has .custom.linux_flavour ["debian", "raspbian", "ubuntu", "android"]) }}
debdiffconf
{{- end }}
```

But I can't figure out how to write a like like that in your template system.

EDIT: I could possibly instead use lookPath and look for dpkg/apt. But I would also still like to know how to test for membership in lists. Especially since the function has is right there in the docs: http://masterminds.github.io/sprig/lists.html

**Comment by bradenhilton** [comment]
Have you tried `if not has ...`?

http://masterminds.github.io/sprig/lists.html#has-musthas

**Reply by VorpalBlade** [reply]
I believe so:

```
{{- if not (has .custom.linux_flavour ["debian", "raspbian", "ubuntu", "android"]) }}
debdiffconf
{{- end }}
```

results in `chezmoi: template: /home/arvid/.local/share/chezmoi/bin/.chezmoiignore:64: unexpected "[" in operand`

**Reply by VorpalBlade** [reply]
This also doesn't work:

```
{{- if not (has .custom.linux_flavour  "debian" "raspbian" "ubuntu" "android" ) }}
debdiffconf
{{- end }}
```

`chezmoi: template: /home/arvid/.local/share/chezmoi/bin/.chezmoiignore:64:12: executing "/home/arvid/.local/share/chezmoi/bin/.chezmoiignore" at <has>: wrong number of args for has: want 2 got 5`

In fact I can't find any formal grammar definition or even *informal* grammar definition anywhere. So I don't even know what a list literal should look like!

**Reply by bradenhilton** [reply, marked as answer]
You probably need something like `if not (has .custom.linux_flavour (list "debian" "raspbian" "ubuntu" "android"))`

**Reply by VorpalBlade** [reply]
Yes that appears to work. Thanks! (The syntax isn't very well documented in my opinion, so that is what I say in my defence.)
